src/plugins/data/server/autocomplete/value_suggestions_route.ts

```diff
@@ -44,24 +44,36 @@ export function registerValueSuggestionsRoute(
       const config = await config$.pipe(first()).toPromise();
       const { field: fieldName, query, filters } = request.body;
       const { index } = request.params;
-      const { client } = context.core.elasticsearch.legacy;
       const signal = getRequestAbortedSignal(request.events.aborted$);
 
       const autocompleteSearchOptions = {
         timeout: `${config.kibana.autocompleteTimeout.asMilliseconds()}ms`,
         terminate_after: config.kibana.autocompleteTerminateAfter.asMilliseconds(),
       };
 
-      const indexPattern = await findIndexPatternById(context.core.savedObjects.client, index);
-
-      const field = indexPattern && getFieldByName(fieldName, indexPattern);
+      const indexPatterns = await context.indexPatterns.find(index, 1);
```

Comment:
I think you want to use `get` instead of `find`. `find` searches by title, `get` uses `id`

Reply:
get uses the `id` fields, and the `index` here is the `title`.